compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/model/types/OrderedTypeVisitor.java

```diff
@@ -0,0 +1,35 @@
+/*
+ *  Copyright (c) 2021, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  WSO2 Inc. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+package org.wso2.ballerinalang.compiler.semantics.model.types;
+
+/**
+ * @param <T> the argument type of the visit methods.
+ * @param <R> the return type of the visit methods.
+ * @since 2.0.0
+ */
+public interface OrderedTypeVisitor<T, R> {
```

Comment:
Can't we implement `org.wso2.ballerinalang.compiler.semantics.model.types.BTypeVisitor` instead of introducing a new interface?

Reply:
We can implement the BTypeVisitor too and override all the visit methods. Introduced the new interface to handle just the types that needed to be handled for checking whether they belong to the same ordered type. What would be the better approach?

Reply:
I guess one benefit of implementing BTypeVisitor is that in case a new type is ever added, a change to BTypeVisitor will force ordered type visitor to consciously make a decision regarding whether or not any changes are needed. But I guess the chances of that happening are pretty slim.

One other advantage is that we don't have to add new visit methods to the types for this visitor.

Reply:
Implemented the `org.wso2.ballerinalang.compiler.semantics.model.types.BTypeVisitor` and removed the new interface via 9f9571c